### DISCUSSION TITLE
Service.prepare(...) doesn't save the given Permission

### DISCUSSION THREAD

**Discussion by JHarzenetter** [schwabe/ics-openvpn#1683] [Q&A] [answered]
Hey Guys,

i have the following problem. I'm using the AIDL-API from this App to controll the VPN directly. But I get asked every time if trust my App to controll OpenVPN for Android. In the exmaple app i get asked once as it should be.

This is my calling CodeFragment (Kotlin). The Code is from the example App which is provided [here](https://github.com/schwabe/ics-openvpn/blob/4ef3bb1f0b027e86e8ef7a62449c0419b0edaae5/remoteExample/src/main/java/de/blinkt/openvpn/remote/MainFragment.java#L151):
```
try {
  val intent = mService!!.prepare(--App Name--)
  if (intent != null) {
    permissionLauncher.launch(intent)
  } else {
    onActResult(ActivityResult(RESULT_OK, null))
  }
} catch (e: RemoteException) {
  e.printStackTrace();
}
```

Does anyone have an idea whats the problem here? Am i missing something?

**Comment by schwabe** [comment, marked as answer]
Does this also happen for the example app? Also in your previous revision you had not your package name as argument for the `.prepare` argument. IIrc you need your package name instead.

**Reply by JHarzenetter** [reply]
The package Name does the deal
